From 8563926d31049936c1dbbba12fb23bcd615eaab4 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Thu, 28 Jan 2021 14:51:50 +0200
Subject: [PATCH] [SW-20154] add capability to report additional status

There are 8 SW_STATUS registers available in PSOC
SECURITY to report ROM Status. Use two of them to
report maximum status information in the ROM.
Rest of the six registers will be used in by eROM
SWs to report status. This way, the STATUS registers
will be mutually exclusive.

Change-Id: I8bd8de66a7fb8c36a57552aad7f489083d39330d
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 6465394..ec00cc4 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -26,7 +26,7 @@
 #define ARCBOOT_BRINGUP_ERROR		-1
 #define ARCBOOT_BRINGUP_OK		1
 
-void arcboot_rom_status_update(enum arcboot_rom_progress progress)
+__attribute__((weak)) void arcboot_rom_status_update(enum arcboot_rom_progress progress)
 {
 	uint32_t reg;
 
diff --git a/gaudi2_arc/common.c b/gaudi2_arc/common.c
index 81f8797..af86bf7 100644
--- a/gaudi2_arc/common.c
+++ b/gaudi2_arc/common.c
@@ -14,6 +14,20 @@
 #include <psoc_reset_conf_masks.h>
 #include <psoc_reset_conf_regs.h>
 
+#define ARCBOOT_STATUS_REG_SZ_BITS	32
+
+void arcboot_rom_status_update(enum arcboot_rom_progress progress)
+{
+	uint32_t reg, offset;
+
+	offset = progress / ARCBOOT_STATUS_REG_SZ_BITS;
+	progress = progress % ARCBOOT_STATUS_REG_SZ_BITS;
+
+	reg = regmap_readl((mmPSOC_SECURITY_SW_STATUS_0 + offset * 4));
+	reg |= BIT(progress);
+	regmap_writel(reg, (mmPSOC_SECURITY_SW_STATUS_0 + offset  * 4));
+}
+
 uint32_t get_efuse_read_reg(void)
 {
 	return mmPSOC_EFUSE_READ_CTRL;
